tests/sagemaker-tests/test_sagemaker_args.py

```diff
@@ -1,62 +1,88 @@
 
+
 def test_empty_value():
-    args = ['--empty-arg', '--next-arg', 'value']
-    expected = {'empty-arg': '', 'next-arg': 'value'}
+    args = ["--empty-arg", "--next-arg", "value"]
+    expected = {"empty-arg": "", "next-arg": "value"}
     assert dict(parse_unknown_gs_args(args)) == expected
 
+
 def test_no_args():
     args = []
     result = parse_unknown_gs_args(args)
     assert len(result) == 0
 
+
 def test_only_flags():
-    args = ['--flag1', '--flag2', '--flag3']
-    expected = {'flag1': '', 'flag2': '', 'flag3': ''}
+    args = ["--flag1", "--flag2", "--flag3"]
+    expected = {"flag1": "", "flag2": "", "flag3": ""}
     assert dict(parse_unknown_gs_args(args)) == expected
 
+
 def test_mixed_args():
-    args = ['--arg1', 'value1', '--flag', '--arg2', 'value2a', 'value2b']
-    expected = {'arg1': 'value1', 'flag': '', 'arg2': 'value2a value2b'}
+    args = ["--arg1", "value1", "--flag", "--arg2", "value2a", "value2b"]
+    expected = {"arg1": "value1", "flag": "", "arg2": "value2a value2b"}
     assert dict(parse_unknown_gs_args(args)) == expected
 
+
 def test_quoted_args():
-    args = ['--arg', '"quoted value"', '--another-arg', "'single quoted'"]
-    expected = {'arg': '"quoted value"', 'another-arg': "'single quoted'"}
-    assert dict(parse_unknown_gs_args(args)) == expected
-
-@pytest.mark.parametrize("input_args, expected_output", [
-    (['--num', '1', '2', '3', '--str', 'a', 'b', 'c'],
-     {'num': '1 2 3', 'str': 'a b c'}),
-    (['--mixed', 'str', '123', 'true', '--flag'],
-     {'mixed': 'str 123 true', 'flag': ''}),
-    (['--empty', '', '--zero', '0', '--false', 'false'],
-     {'empty': '', 'zero': '0', 'false': 'false'}),
-    (['--special-chars', '!@#$%^&*()'],
-     {'special-chars': '!@#$%^&*()'}),
-    (['--unicode', 'こんにちは', 'world'],
-     {'unicode': 'こんにちは world'}),
-])
+    args = ["--arg", '"quoted value"', "--another-arg", "'single quoted'"]
+    expected = {"arg": '"quoted value"', "another-arg": "'single quoted'"}
+    assert dict(parse_unknown_gs_args(args)) == expected
+
+
+@pytest.mark.parametrize(
+    "input_args, expected_output",
+    [
+        (["--special-chars", "!@#$%^&*()"], {"special-chars": "!@#$%^&*()"}),
+        (["--unicode", "こんにちは", "world"], {"unicode": "こんにちは world"}),
+    ],
+)
 def test_parse_unknown_gs_args_parametrized(input_args, expected_output):
     assert dict(parse_unknown_gs_args(input_args)) == expected_output
+
+
+def test_parse_single_string():
+    """Happens when GS args are passed in quoted:
+    ``python launch_*.py --launch-arg 1 '--gs-arg1 2 --gs-arg2 3'``
+    """
+    args = ["--num-epochs 1 --use-graphbolt true"]
+    expected = {"num-epochs": "1", "use-graphbolt": "true"}
+    assert dict(parse_unknown_gs_args(args)) == expected
+
+
+def test_parse_complex_string_with_quotes():
+    args = [
+        (
+            "--target-etype query,clicks,asin query,search,asin "
+            "--feat-name ntype0:feat0 ntype1:feat1"
```

Comment:
There is no quotes in the input string.

Reply:
The quoting refers to the CLI call in bash, like `python launch_*.py --instance-type [ml...] '--feat-name ntype0:feat0 ntype1:feat1'`

This happened e.g. in my training automation script where my call was


```bash
TRAIN_YAML_S3="s3://my-bucket/yaml-files/acm/acm_nc.yaml"
GSF_ARGS="--num-epochs 1 --use-graphbolt true"

python3 $SCRIPT_DIR/launch/launch_train.py \
    --entry-point run/train_entry.py \
    --yaml-s3 ${TRAIN_YAML_S3} \
    # Other launch args...
    "${GSF_ARGS}" # Note the double quotes
```

Generally it's good practice to use the double quotes in bash vars where we might have spaces and special characters: https://www.shellcheck.net/wiki/SC2086 so we should support this case.